### DISCUSSION TITLE
When mozilla tts will be in firefox

### DISCUSSION THREAD

**Discussion by JRMeyer** [coqui-ai/TTS#324] [Archived]
<USER>>>> hkalbasi<br/> <DATE> [December 16, 2020, 7:55am]<br/> <br/> Currently, Firefox is using espeak for speech synthesis api and reader<br/> mode narrator, Which is much worse than mozilla tts (In term of result<br/> quality). Is there a specific time for replace espeak with mozilla tts<br/> in supported languages? Or an option for choosing between them in a user<br/> friendly way?<br/> <br/> [This is an archived TTS discussion thread from discourse.mozilla.org/t/when-mozilla-tts-will-be-in-firefox]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> othiele<br/> <DATE> [December 16, 2020, 8:34am]<br/> <br/> I don't know if erogol has time to answer that. Mozilla has decided not<br/> to continue putting much effort into voice. They put people on other<br/> tasks. It is therefore unlikely that any open TTS tech will be included<br/> in Mozilla anytime soon. Sorry.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> hkalbasi<br/> <DATE> [December 16, 2020, 9:37am]<br/> <br/> How bad. Why did they make this decision now that the tree has borne<br/> fruit? Even if they replace espeak with mozilla tts now and stop tts<br/> development completely, I think it will be a definite improvement in<br/> firefox.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> othiele<br/> <DATE> [December 16, 2020, 10:29am]<br/> <br/> ![](../../user_avatar/discourse.mozilla.org/mitchell/40/22219_2.png){.avatar<br/> community](https://discourse.mozilla.org/t/dear-mozilla-community/65546)<br/> .clear-badge style='color: #FFFFFF' drop-close='true'<br/> .box}<br/> <br/> > Dear Mozilla community Today Mozilla announced some significant<br/> > changes. The tl;dr is that Mozilla Corporation is undergoing a<br/> > significant reorganization, as well as a reduction in the number of<br/> > employees of about 25%. The best explanation for these changes is in<br/> > our [blog<br/> > post](https://blog.mozilla.org/blog/2020/08/11/changing-world-changing-mozilla/)<br/> > and in the [message to<br/> > employees](https://blog.mozilla.org/wp-content/uploads/2020/08/Message-to-Employees-Change-in-Difficult-Times.pdf)<br/> > that we included in the post. There is important information and<br/> > context in these messages, and I believe it will be very helpful to<br/> > read these when you have a moment As part of the chang...<br/> <br/> [Archived Post]<br/>

**Comment by ghost** [comment]
Commenting here as this issue appears quite high in Google results for "firefox coqui tts."

Firefox doesn't use espeak directly, it rather uses [speech dispatcher](https://github.com/brailcom/speechd/). It's possible (but not simple) for this to use Coqui TTS via the "generic" module, see: https://github.com/coqui-ai/TTS/discussions/2583

**Comment by kdavis-coqui** [comment]
This is a decision for Mozilla not Coqui. Coqui has no influence on Mozilla's product direction.

**Comment by genodeftest** [comment]
There is also [Local TTS](https://addons.mozilla.org/en-US/firefox/addon/local-tts/?utm_source=addons.mozilla.org&utm_medium=referral&utm_content=search), a firefox extension that claims to be able to use a local coqui server. Has anyone got this to work yet?

It does not describe how to run a local coqui server.
